vmm/src/memory_manager.rs

```diff
@@ -1477,9 +1462,121 @@ impl MemoryManager {
                 .map_err(Error::ApplyNumaPolicy)?;
         }
 
+        // Prefault the region if needed, in parallel.
+        if prefault {
+            let page_size =
+                Self::get_prefault_align_size(backing_file, hugepages, hugepage_size)? as usize;
+            let num_pages = size / page_size;
```

Comment:
What if `size` is not `page_size` aligned? I know this is mostly a user mis-configuration, but I think we should generate a `warn!()` for such case.

Reply:
If we don't already have a check for the memory size being a multiple of the page size then that is something that should be solved at an earlier time.

Reply:
As @rbradford suggested there is an earlier check [here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/3042a0baf3829e9ad6fd8a6d08b0b5fdc55bc42f/vmm/src/memory_manager.rs#L541) for proper alignment of the memory size and the page size in `create_memory_regions_from_zones` before `create_ram_region` is called:
```
        let mut zone_align_size = memory_zone_get_align_size(zone)?;
...
        if !is_aligned(zone.size, zone_align_size) {
            return Err(Error::MisalignedMemorySize);
        }
```
However, there is another path to `create_ram_region` through `add_ram_region`, which itself is guarded by a check in `hotplug_ram_region` [here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/3042a0baf3829e9ad6fd8a6d08b0b5fdc55bc42f/vmm/src/memory_manager.rs#L1571) that the size is a multiple of 128 MiB, which may not be properly aligned with 1 GiB hugepages. 

@likebreath I've added a `warn!()` to address this.